src/library_fs.js

```diff
@@ -26,7 +26,55 @@ mergeInto(LibraryManager.library, {
     addAtInit('if (!Module["noFSInit"] && !FS.init.initialized) FS.init();');
     addAtMain('FS.ignorePermissions = false;');
     addAtExit('FS.quit();');
-    return 'FS.staticInit();' +
+    // We must statically create FS.FSNode here so that it is created in a manner
+    // that is visible to Closure compiler. That lets us use type annotations for
+    // Closure to the "this" pointer in various node creation functions.
+    return 'var FSNode = /** @constructor */ function(parent, name, mode, rdev) {\n'+
```

Comment:
I think you can use JS multiline strings for stuff like this, a single backtick on both sides,
```
`var FSNode = ...
...
...`
```

Reply:
Another option:
```
return getBody(function() {
  var FSNode = ..
});
```
and we add `getBody()` which returns the body of a JS function, as a string. We'd need to verify that doesn't affect comments though.